ppr-ui/src/components/parties/composables/useSecuredParty.ts

```diff
@@ -96,21 +97,36 @@ export const useSecuredParty = (props, context) => {
   }
 
   const hasMatchingSecuredParty = (addedParty: PartyIF): boolean => {
-    // store state without newly added party
+    // store state without newly added party.
     const parties = cloneDeep(getAddSecuredPartiesAndDebtors.value.securedParties)
     if (localState.partyType === SecuredPartyTypes.INDIVIDUAL) {
       return parties.some(party =>
-        isEqual(party.personName, addedParty.personName) &&
-        isEqual(party.address, addedParty.address)
+        isObjectEqual(party.personName, addedParty.personName) &&
+        isObjectEqual(party.address, addedParty.address)
       )
     } else {
       return parties.some(party =>
-        party.businessName === addedParty.businessName &&
-        isEqual(compact(party.address), compact(addedParty.address))
+        party.businessName?.toUpperCase() === addedParty.businessName?.toUpperCase() &&
+        isObjectEqual(party.address, addedParty.address)
       )
     }
   }
 
+  const isObjectEqual = (object1: any, object2: any): boolean => {
+    const workObject1 = Object.create(object1)
+    const workObject2 = Object.create(object2)
+    const addressEqual = isEqual(normalizeObject(workObject1), normalizeObject(workObject2))
+    return addressEqual
+  }
+
+  const normalizeObject = (convertObject: any): any => {
+    convertObject = omitBy(convertObject, overSome([isNil, isNaN, isEmpty]))
+    Object.entries(convertObject).forEach(([key, value]) => {
+      if (typeof value === 'string') convertObject[key] = value?.toUpperCase().replaceAll(' ', '').trim()
+    })
+    return convertObject
+  }
```

Comment:
We have a helper method called `cleanEmpty()`, this could potentially solve the same issue? 

The only concern here, this is adding in complex custom code, that isn't typed and challenging to follow without looking documentation on all the various helper imports. 

If you feel this is absolutely required, than please put it somewhere more global(utils) and with proper doc bloc comments so others can follow and use again 👍

Reply:
Ill have a look at cleanEmpty...If it does'nt work I will move function to utills..

Reply:
Sounds good, thanks! 

I think if you could type this and add some intuitive doc bloc comments, it would go a long way to make this more intuitive for other developers.

Reply:
cleanEmpty gives the same functionality as omit which is a lodash function..
I do like the idea of moving to utils.

Which import would you suggest... I think this function will be used elsewhere..

Reply:
`validation-helper.ts` seems like a fit?

Reply:
Funny.  That's where I put it... I have checked in the code.